### DISCUSSION TITLE
Is there a way to alphabeticaly order teams in the left panel?

### DISCUSSION THREAD

**Discussion by evanchaoli** [concourse/concourse#5412] [General]
A pipeline admin maybe in a lot of teams, it's hard to locate a team given there is a long team list in the left panel. So is there a way to show teams in alphabetical order in the left panel on the GUI?

**Comment by skreddy6673** [comment]
Yup, this is so helpful when you logged in as admin.

**Comment by vito** [comment]
Hmm, we'll be working on a 'favoriting pipelines' feature soon. Don't know if that would help though. Ordering alphabetically seems sensible - perhaps with teams you're actually a member of listed first. I'm actually not sure what the current ordering is based on.

@matthewpereira Any thoughts on this?

**Reply by matthewpereira** [reply]
I think alphabetical sorting makes a lot of sense. 👍 We could eventually add a 'sort by recency' or 'sort by number of pipelines in each team' or a number of other options too, with ascending and descending variations.

I'm hoping that some of the view enhancements that are on the roadmap will help, but confusion over how Concourse sorts things has come up a number of times - some bandaid solution to make it more predictable would be great.

**Comment by jamieklassen** [comment]
I can comment what currently determines the team order. We can see here:
https://github.com/concourse/concourse/blob/3b487b5b48a4ff6039330cd98b65c3f79b6eb80d/web/elm/src/SideBar/SideBar.elm#L147

that the view function makes use of `List.Extra.gatherEqualsBy`, and from its documentation:

https://github.com/elm-community/list-extra/blob/4a1b8674db0ddbb84294de01807b373d7a5b06bd/src/List/Extra.elm#L1927-L1928

So the real source of truth is however the backend is sorting the items in the `ListAllPipelines` response:

https://github.com/concourse/concourse/blob/3b487b5b48a4ff6039330cd98b65c3f79b6eb80d/atc/api/pipelineserver/list_all.go#L22

which, in the case of a super-admin, is `pipelineFactory.AllPipelines`:

https://github.com/concourse/concourse/blob/3b487b5b48a4ff6039330cd98b65c3f79b6eb80d/atc/db/pipeline_factory.go#L75

so the ordering is by team id, and then by the `ordering` column. Since `team_id` is a foreign key:

https://github.com/concourse/concourse/blob/3b487b5b48a4ff6039330cd98b65c3f79b6eb80d/atc/db/migration/migrations/1510262030_initial_schema.up.sql#L391

to an auto-incrementing ID field:

https://github.com/concourse/concourse/blob/3b487b5b48a4ff6039330cd98b65c3f79b6eb80d/atc/db/migration/migrations/1510262030_initial_schema.up.sql#L549-L570

This really means that teams in the sidebar will appear in the order they were created.

**Comment by skreddy6673** [comment]
I just checked on my end and it's exactly how the teams are displayed.

**Comment by evanchaoli** [comment]
@vito Should this be converted to an issue?

**Reply by vito** [reply]
With our new community structure, converting it to an issue would mean it goes on our backlog/roadmap. We've validated the idea, but we don't necessarily have bandwidth for it at this point, so I don't want to set an expectation that this will happen through us implementing it and make everyone wait on us since it's not likely to be a high priority.

That being said, I do see a need for a sort of "Help Wanted" category for things like this, which would previously be done through issues. Once Discussions supports categories we could make one for it. Hopefully that's sometime soon. 🙏

**Reply by evanchaoli** [reply]
Okey, let me submit a PR.

**Comment by evanchaoli** [comment]
My PR https://github.com/concourse/concourse/pull/5622 has been merged.
